framework/source/class/qx/util/DynamicScriptLoader.js

```diff
@@ -0,0 +1,241 @@
+ *      src = '.src';
+ *      min = '';
+ *    }
+ *
+ *    // initialize the script loading
+ *    var dynLoader = new qx.util.DynamicScriptLoader();
+ *
+ *
+ *    dynLoader.addListenerOnce('ready',function(e){
+ *      console.log("all scripts have been loaded!");
+ *    });
+ *
+ *    dynLoader.addListener('failed',function(e){
+ *      var data = e.getData();
+ *      console.log("failed to load "+data.script);
+ *    });
+ *
+ *    dynLoader.load([
+ *        "myapp/jquery/jquery"+min+".js",
+ *        "myapp/highcharts/highcharts"+src+".js",
+ *        "myapp/highcharts/highcharts-more"+src+".js",
+ *        "myapp/highcharts/highcharts-modifications.js"
+ *    ]);
+ *    
+ * </code>
+ */
+qx.Class.define("qx.util.DynamicScriptLoader", {
+  extend: qx.core.Object,
+
+  /**
+   * Initialize the state hashes if not yet done
+   */
+
+  construct: function () {
+    this.base(arguments);
+    this.__QUEUE = [];
+    this.__LOADING = false;
+  },
+
+  /*
+  *****************************************************************************
+     EVENTS
+  *****************************************************************************
+  */
+
+  events: {
+    /**
+     * fired when a script is loaded successfully. The data contains 'script' and 'status' keys.
+     */
+    loaded: 'qx.event.type.Data',
+
+    /**
+     * fired when a specific script fails loading.  The data contains 'script' and 'status' keys.
+     */
+    failed: 'qx.event.type.Data',
+
+    /**
+     * fired when all given scripts are loaded, each time loadScriptsDynamic is called.
+     */
+    ready: 'qx.event.type.Event'
+  },
+
+  statics: {
+    /**
+     * Map of scripts being added at the present time. Key is script name; value is instance of this class which
+     * is loading it.
+     */
+    __IN_PROGRESS: {},
+     /**
+      * Map of scripts that have fully loaded. Key is script name; value is true
+      */
+    __LOADED: {}
+  },
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
+
+
+  members: {
+
+    __QUEUE: null,
+    __LOADING: null,
+
+
+    /**
+     * Load scripts dynamically, typically used in the instance constructor
+     *
+     * @param codeArr {Array} an array with the uri names of the scripts
+     *
+     */
+    load: function(codeArr) {
+      codeArr.forEach(function(script) {
+         this.__QUEUE.push(script);
+      },this);
+      this.__loadScripts();
+    },
+
+
+    /**
+     * Chain loading scripts.
+     *
+     * Recursively called until the array of scripts is consumed
+     *
+     * @param codeArr {Array} an array with the uri names of the scripts
+     */
+    __loadScripts: function() {
+      var cl = qx.util.DynamicScriptLoader;
+      var script;
+      var dynLoader;
+      var id1, id2;
+      var uri;
+      var loader;
+
+      if (this.__LOADING === true) {
+        return;
```

Comment:
Maybe the ```load``` method should test ```if(this.__LOADING)``` and return false if this is the case, to indicate that the loader is currently not accepting another request. And of course return true to signal loading was accepted and started.